### PR TITLE
fix(ai): align toolsToModelTools with core AI SDK's prepareToolsAndToolChoice

### PR DESCRIPTION
**Pull request by iNishant**
## Summary

`@workflow/ai`'s `toolsToModelTools` only forwards `type`, `name`, `description`, and `inputSchema` when building `LanguageModelV3FunctionTool` objects. The core AI SDK's `prepareToolsAndToolChoice` forwards three additional properties:

- **`strict`** — enables strict schema validation on providers that support it (e.g. OpenAI)
- **`inputExamples`** — optional examples for the language model
- **`providerOptions`** — provider-specific tool options

This means tools using these properties lose them when run through `DurableAgent`.

This PR aligns `toolsToModelTools` with the core AI SDK's implementation in `ai/src/prompt/prepare-tools-and-tool-choice.ts`.

## Changes

**`packages/ai/src/agent/tools-to-model-tools.ts`**

Before:
```ts
export function toolsToModelTools(
  tools: ToolSet
): LanguageModelV2FunctionTool[] {
  return Object.entries(tools).map(([name, tool]) => ({
    type: 'function',
    name,
    description: tool.description,
    inputSchema: asSchema(tool.inputSchema).jsonSchema,
  }));
}
```

After:
```ts
export async function toolsToModelTools(
  tools: ToolSet
): Promise<LanguageModelV3FunctionTool[]> {
  return Promise.all(
    Object.entries(tools).map(async ([name, tool]) => ({
      type: 'function' as const,
      name,
      description: tool.description,
      inputSchema: await asSchema(tool.inputSchema).jsonSchema,
      ...(tool.inputExamples != null
        ? { inputExamples: tool.inputExamples }
        : {}),
      providerOptions: tool.providerOptions,
      ...(tool.strict != null ? { strict: tool.strict } : {}),
    }))
  );
}
```

Also updates the return type from `LanguageModelV2FunctionTool` to `LanguageModelV3FunctionTool` and makes the function `async` to match the upstream signature change.

### CODE REVIEW

**Review by VaguelySerious — COMMENTED** [review]
AI review: blocking issues found

File: packages/ai/src/agent/tools-to-model-tools.ts [hunk 1]
```diff
@@ -1,6 +1,8 @@
 import type { LanguageModelV3FunctionTool } from '@ai-sdk/provider';
```
Comment (VaguelySerious) [inline packages/ai/src/agent/tools-to-model-tools.ts]: ### AI Review: Blocking

Missing changeset. Per repo policy, every PR requires a changeset. A patch changeset for `@workflow/ai` should be added, e.g.:

```
---
"@workflow/ai": patch
---

Forward `strict`, `inputExamples`, and `providerOptions` tool properties to language model providers
```

File: packages/ai/src/agent/tools-to-model-tools.ts [hunk 2]
```diff
@@ -10,6 +12,11 @@ export async function toolsToModelTools(
       name,
       description: tool.description,
       inputSchema: await asSchema(tool.inputSchema).jsonSchema,
+      ...(tool.inputExamples != null
+        ? { inputExamples: tool.inputExamples }
+        : {}),
+      providerOptions: tool.providerOptions,
+      ...(tool.strict != null ? { strict: tool.strict } : {}),
```
Comment (VaguelySerious) [inline packages/ai/src/agent/tools-to-model-tools.ts]: ### AI Review: Note

This PR aligns `toolsToModelTools` with the AI SDK for `function`/`dynamic`/`undefined` tool types, but the AI SDK's `prepareToolsAndToolChoice` also handles `type: 'provider'` tools (e.g. OpenAI's `web_search`, `code_interpreter`, `file_search`) by producing a `LanguageModelV3ProviderTool` with `{ type: 'provider', name, id, args }`.

Currently, a provider-type tool passed through `toolsToModelTools` would produce a malformed `LanguageModelV3FunctionTool`. The `ToolSet` type does include `type: 'provider'` tools, and the downstream `doStreamStep` already accepts `LanguageModelV3ProviderTool` in its `tools` parameter.

This is a pre-existing gap (not a regression from this PR), but since the stated goal is to "align with the core AI SDK's implementation," it may be worth addressing here.

File: packages/ai/src/agent/tools-to-model-tools.ts [hunk 3]
```diff
@@ -1,6 +1,8 @@
 import type { LanguageModelV3FunctionTool } from '@ai-sdk/provider';
 import { asSchema, type ToolSet } from 'ai';
 
+// Mirrors the tool→LanguageModelV3FunctionTool mapping in the core AI SDK's
+// prepareToolsAndToolChoice (ai/src/prompt/prepare-tools-and-tool-choice.ts).
 export async function toolsToModelTools(
```
Comment (VaguelySerious) [inline packages/ai/src/agent/tools-to-model-tools.ts:10]: ### AI Review: Note

There are no existing tests for `tools-to-model-tools.ts` and this PR doesn't add any. Validation testing confirmed all three properties are correctly forwarded (including edge cases like `strict: false` being preserved, and `inputExamples`/`strict` keys being absent when not set). Consider adding unit tests to prevent regressions if the AI SDK changes its tool type shape in future versions.

**Comment by iNishant** [comment]
@VaguelySerious good to go?

**Comment by VaguelySerious** [comment]
@iNishant Thanks for the reminder, yes. DCO is failing, could you ensure that passes? Will merge after

**Review by VaguelySerious — APPROVED** [review]

**Comment by iNishant** [comment]
@VaguelySerious did the DCO thing.

**Comment by VaguelySerious** [comment]
Thanks for your patience!
